src/main/java/org/generationcp/middleware/service/impl/inventory/LotIDGeneratorImpl.java

```diff
@@ -0,0 +1,43 @@
+
+package org.generationcp.middleware.service.impl.inventory;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.pojos.ims.Lot;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.generationcp.middleware.service.api.LotIDGenerator;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.UUID;
+
+@Transactional
+@Service
+public class LotIDGeneratorImpl implements LotIDGenerator {
```

Comment:
@vmaletta any advantage of having an isolated generator instead of embedding it in the services or datamanagers?

Reply:
Removed